### PR TITLE
feat: Add monitoring data gathering

### PR DESCRIPTION
**Pull request by jhutar**

### CODE REVIEW

**Review by pmacik — CHANGES_REQUESTED** [review]
@jhutar Nice, good job!

There are some corner cases that need to be fixed, though - PTAL 
https://github.com/pmacik/backstage-performance/commit/21a5edc8c9a6f5c26d77e9730a91b9c7a7504541

**Comment by jhutar** [comment]
Thank @pmacik . Cherry-picked your commit.

File: ci-scripts/rhdh-setup/deploy.sh [hunk 1]
```diff
@@ -47,11 +47,73 @@ backstage_install() {
     $clin wait --for=condition=Ready pod -l=app.kubernetes.io/name=developer-hub --timeout=300s
 }
 
+setup_monitoring() {
+    echo "Enabling user workload monitoring"
+    rm -f config.yaml
+    if oc -n openshift-monitoring get cm cluster-monitoring-config; then
+        oc -n openshift-monitoring extract configmap/cluster-monitoring-config --to=. --keys=config.yaml
+        sed -i 's/enableUserWorkload:.*/enableUserWorkload: true/g' config.yaml
```
Comment (pmacik) [inline ci-scripts/rhdh-setup/deploy.sh:55]:
```suggestion
        sed -i '/^enableUserWorkload:/d' config.yaml
        echo -e "\nenableUserWorkload: true" >> config.yaml
```

**Comment by jhutar** [comment]
It worked! But looks like test ends only after `oc logs ...` end :-) Let's try again.

/test baseline-pr

**Comment by pmacik** [comment]
> It worked! But looks like test ends only after `oc logs ...` end :-) Let's try again.
> 
> /test baseline-pr

yes, my mistake, sorry

**Comment by jhutar** [comment]
> > It worked! But looks like test ends only after `oc logs ...` end :-) Let's try again.
> > /test baseline-pr
> 
> yes, my mistake, sorry

Not at all! Thanks for all the fixes!

**Pull request merged**